### PR TITLE
Fix/286 react to route change

### LINKED ISSUE #286: Changing entity in url in Data Explorer 2 doesn't trigger routing
### How to Reproduce
Go to master.dev.molgenis.org
click on experimental and click on Collections
Change the eu_bbmri_eric_collections to eu_bbmri_eric_biobanks

### Expected behavior
View is updated to represent new selected entity

### Observed behavior
No change in the frontend

### PR DESCRIPTION
**Pull request by connoratrug**
fix: #286 react to route change 

Closes #286 

- Cleanup router; single route + catch other route path.
- Add route listener to MainView to react to route changes.
   - Remove all onCreated/onMounted testup code ( replaced my listing to route changes).
- Simplfy get data function ( expect meta to be loaded).
- Sync loading of data, first load Setting and Meta, then load data.
- Add comment to filter watchers, these can be removed once filter is part of route.
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] Conventional commits (squash if needed)
- [ ] No warnings during install
- [ ] Updated javascript typing

### CODE REVIEW

File: packages/data-explorer/src/store/mutations.ts [hunk 1]
```diff
@@ -69,8 +62,10 @@ export default {
   setFilterSelection (state: ApplicationState, selections: StringMap) {
     Vue.set(state.filters, 'selections', selections)
   },
-  setFilters (state: ApplicationState, { definition, shown }: { definition: FilterDefinition[], shown: string[] }) {
+  setFilterDefinition (state: ApplicationState, definition: FilterDefinition[]) {
```
Comment (jelmerveen) [inline packages/data-explorer/src/store/mutations.ts:65]: +1 for semantics

File: packages/data-explorer/src/store/state.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import ApplicationState from '@/types/ApplicationState'
 
 const state: ApplicationState = {
   toast: null,
-  tableName: 'root_hospital_patients',
+  tableName: null,
```
Comment (jelmerveen) [inline packages/data-explorer/src/store/state.ts:5]: yes, no more hard-coded table

File: packages/data-explorer/src/store/actions.ts [hunk 3]
```diff
@@ -8,22 +8,37 @@ import * as metaDataService from '@/repository/metaDataService'
...
   fetchCardViewData: async ({ commit, state, getters }: { commit: any, state: ApplicationState, getters: any }) => {
     if (state.tableName === null) {
-      throw new Error('cannot load table data for non string table id')
+      throw new Error('cannot load card data without table name')
     }
 
-    const metaData = await metaDataRepository.fetchMetaDataById(state.tableName)
-    commit('setMetaData', metaData)
-    commit('setFilters', await metaFilterMapper.mapMetaToFilters(metaData))
+    if (state.tableMeta === null) {
```
Comment (jelmerveen) [inline packages/data-explorer/src/store/actions.ts:39]: this also triggers when a user is not logged in

File: packages/data-explorer/src/store/actions.ts [hunk 4]
```diff
@@ -8,22 +8,37 @@ import * as metaDataService from '@/repository/metaDataService'
...
+      throw new Error('cannot load table data without meta data')
```
Comment (jelmerveen) [inline packages/data-explorer/src/store/actions.ts:40]: triggers also when not logged in

File: packages/data-explorer/src/views/SelectLayoutView.vue [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="mt-3 entity-table" v-show="tableData && tableData.items && tableData.items.length > 0 && tableMeta">
+  <div class="mt-3 entity-table" v-if="tableMeta && tableData">
     <div v-if="tableData && tableData.items && tableData.items.length === 0" class="alert alert-warning">
       {{ 'dataexplorer_empty_table' | i18n}}
```
Comment (jelmerveen) [inline packages/data-explorer/src/views/SelectLayoutView.vue:4]: also tirggers when filtering (multifilter) is that intended?

**Comment by jelmerveen** [comment]
refreshing takes a few seconds, which his not always clear.
When deselecting an active filter, they all dissappear, to reappear later again.

When changing entity it takes ~ 1-2 seconds for the view to update.

File: packages/data-explorer/src/store/actions.ts (hunk 4 above)
Comment (connoratrug) [reply packages/data-explorer/src/store/actions.ts:40]: works as intented

**Comment by connoratrug** [comment]
> refreshing takes a few seconds, which his not always clear.
 'which his not always clear' not part of the PO
> When deselecting an active filter, they all dissappear, to reappear later again.
> 
> When changing entity it takes ~ 1-2 seconds for the view to update.

File: packages/data-explorer/src/views/SelectLayoutView.vue (hunk 5 above)
Comment (connoratrug) [reply packages/data-explorer/src/views/SelectLayoutView.vue:4]: yes same as master

**Comment by jelmerveen** [comment]
first i go to:
https://preview-frontend-pr-295-3.dev.molgenis.org/@molgenis-experimental/data-explorer/dist/index.html#/eu_bbmri_eric_collections

in console i get record.order_of_magnitude undefined [this is a custom card]

Now clear the log

then change url to:
https://preview-frontend-pr-295-3.dev.molgenis.org/@molgenis-experimental/data-explorer/dist/index.html#/quest_nillable_Books

open console.log
same errors appear.

Which is weird, because we navigate away from the custom card but it seems like it's loading custom card first, then trigger route.

**Comment by connoratrug** [comment]
> first i go to:
> https://preview-frontend-pr-295-3.dev.molgenis.org/@molgenis-experimental/data-explorer/dist/index.html#/eu_bbmri_eric_collections
> 
> in console i get record.order_of_magnitude undefined [this is a custom card]
> 
> Now clear the log
> 
> then change url to:
> https://preview-frontend-pr-295-3.dev.molgenis.org/@molgenis-experimental/data-explorer/dist/index.html#/quest_nillable_Books
> 
> open console.log
> same errors appear.
> 
> Which is weird, because we navigate away from the custom card but it seems like it's loading custom card first, then trigger route.

Your preview is outdated, please use latest pre-view for up to date code

**Comment by jelmerveen** [comment]
> > first i go to:
> > https://preview-frontend-pr-295-3.dev.molgenis.org/@molgenis-experimental/data-explorer/dist/index.html#/eu_bbmri_eric_collections
> > in console i get record.order_of_magnitude undefined [this is a custom card]
> > Now clear the log
> > then change url to:
> > https://preview-frontend-pr-295-3.dev.molgenis.org/@molgenis-experimental/data-explorer/dist/index.html#/quest_nillable_Books
> > open console.log
> > same errors appear.
> > Which is weird, because we navigate away from the custom card but it seems like it's loading custom card first, then trigger route.
> 
> Your preview is outdated, please use latest pre-view for up to date code

I pulled the latest commits to local, so it should be up-to-date

**Pull request merged**